org.adempiere.pipo/src/org/adempiere/pipo2/PoFiller.java

```diff
@@ -338,13 +379,13 @@ public int setTableReference(String qName) {
 					setInteger(qName);
 				} else if (info.getColumnClass(index) == Timestamp.class) {
 					setTimestamp(qName);
-				}else if(DisplayType.TextLong == info.getColumnDisplayType(index)) {// export column from system have type is normal string, but import to system have this column but type is textlong (mean blob)
-					if (getStringValue (qName) != null && !isBlobOnPackinFile(qName)) {
+				}else if(DisplayType.isText(info.getColumnDisplayType(index))) {// data is text then test to see on 2pack can get text to setter or not
+					if (getTreamentType(qName) == TagValueType.TEXT)

```

Comment:
Hi @hieplq - one doubt here, I see the getTreatmentType has a possible return of BINARY when the column is named BinaryData, but the if here ignores that case, 

is that on purpose?  wouldn't be better then to use a simple else for BINARY and CLOB cases?

Reply:
in case it return BINARY mean we try to import a binary file (image, attach file) on 2pack into string column
it's mismatch just ignore this case

maybe a warning log?

Reply:
Is that check necessary?

I haven't tested, but I mean, what if somebody creates a column named BinaryData and set it as String?  or as a CLOB?

I remember there is something special about that name in core, because there is the Test.BinaryData column, but not sure how would that work in other tables.  Anyways your change is to validate "if(DisplayType.isText" - so, I think you can treat it as String or CLOB (I think both, Binary and CLOB are saved in zip file)

Reply:
check binary to void packin binary (on source database) on 2pack into a string column (on target database)
check depend to columnName = "BinaryData" is weak, but at moment all core table use that name for binary column
i will improve this point, can use data type (byte[] or String) on Element.contents to detect binary type

logic is:
1. attachment save to zip (because we attach many file per record)
2. normal binay (image) save to .dat file
3. clob save to .date file

data type on Element.contents is "string" for clob, "byte[]" for other

Reply:
OK, as you describe it then maybe a warning is fine - it would be a wrong configuration to have a BinaryData column with String/Text content.